server/sms/sms_gateway.go

```diff
@@ -188,43 +160,34 @@ func (g *gateway) proxyLoop() (*protocol.Message, error) {
 			}
 
 			err := g.send(receivedMsg)
-			if err != nil {
-				return receivedMsg, err
+			if err != nil && err == ErrRetryFailed {
+				// THIS MAY BE BLOCKING.Maybe not a good idea.
+				for err2 := g.SetLastSentID(receivedMsg.ID); err2 != nil; {
+					g.logger.WithField("error", err2.Error()).Error("Error setting last ID.Retrying")
+					time.Sleep(100 * time.Millisecond)
```

Comment:
Bigger sleep possible. Like 10 seconds or 1 minute. 
If this problem appears it will take a while to fix.